fluent-bundle/src/lib.rs

```diff
@@ -82,14 +82,49 @@
 #[macro_use]
 extern crate rental;
 
+use intl_memoizer::{IntlLangMemoizer, Memoizable};
+use std::cell::RefCell;
+use unic_langid::LanguageIdentifier;
+
 mod bundle;
+pub mod concurrent;
 mod entry;
 mod errors;
+pub mod memoizer;
 pub mod resolve;
 mod resource;
 pub mod types;
 
-pub use bundle::{FluentArgs, FluentBundle, FluentMessage};
+pub use bundle::{FluentArgs, FluentMessage};
 pub use errors::FluentError;
 pub use resource::FluentResource;
 pub use types::FluentValue;
+
+pub type Memoizer = RefCell<IntlLangMemoizer>;
+
+pub type FluentBundle<R> = bundle::FluentBundleBase<R, Memoizer>;
+
+impl memoizer::MemoizerKind for Memoizer {
+    fn new(lang: LanguageIdentifier) -> Self
+    where
+        Self: Sized,
+    {
+        RefCell::new(IntlLangMemoizer::new(lang))
+    }
+
+    fn with_try_get<I, R, U>(&self, args: I::Args, cb: U) -> Result<R, ()>
```

Comment:
We should document this such that it is considered an error to recursively access the memoizer within the callback.

Reply:
Also we can just remove `with_try_get` and have everyone else use `.lock().try_get()` or `.borrow().try_get()`, though this may make it harder to switch to parking lot.

Reply:
Thanks, I'll look into that in #165!